[Melvan-Chan] iP

Melvan-Chan opened the pull request:
# Veronica  
*"She who brings the victory."*  

Veronica is your personal task manager, designed to help you stay organized effortlessly. It’s:  

- **Text-based** for quick and efficient task management.
- **Easy to use** with simple commands.
- **Fast and reliable**, so you spend less time managing and more time doing.

> [!IMPORTANT]
### Features:  
- [ ] **Task management** – Add, list, mark, and remove tasks  ✅
- [ ] **Deadlines & Events** – Set due dates and time-bound events✅
- [ ] **Task search** – Find tasks quickly with keywords✅
> 🚀 **More enhancements coming soon!**  

> [!TIP]
### How to get started:  
1. **Download** it from [here](https://github.com/Melvan-Chan/ip/releases/tag/A-Jar).  
2. **Run** the program with a double-click.  
3. **Add tasks** like to-dos, deadlines, and events.  
4. **Let Veronica keep track** while you focus on what matters.

Commit:
Add Gradle support

Eclipse-Dominator committed:
Bump gradle and lib version

aureliony committed:
build.gradle: Prevent generating a second JAR file

In build.gradle, the dependencies on distZip and/or distTar causes
the shadowJar task to generate a second JAR file for which the
mainClass.set("seedu.duke.Duke") does not take effect.
Hence, this additional JAR file cannot be run.
For this product, there is no need to generate a second JAR file
to begin with.

Let's remove this dependency from the build.gradle to prevent the
shadowJar task from generating the extra JAR file.

Melvan-Chan committed:
added fruits.txt

Melvan-Chan committed:
replaced banana's with BIG berries

Melvan-Chan committed:
replaced cherries with BIG chocolate

Melvan-Chan committed:
Rename, Greet, Exit

Melvan-Chan committed:
Echo

Melvan-Chan committed:
Updated greet text

Melvan-Chan committed:
Add, List

Melvan-Chan committed:
Mark as Done

Melvan-Chan committed:
ToDos, Events, Deadlines

Melvan-Chan committed:
Update code to Egyptian style and add test files.

Melvan-Chan committed:
Add error handlers

Melvan-Chan committed:
Add task removal functionality

Melvan-Chan committed:
Remove unused variable

Melvan-Chan committed:
Add enum to task system

Melvan-Chan committed:
Update list system to handle empty list

namprice227 commented on src/main/java/Veronica.java:
```diff
@@ -0,0 +1,152 @@
+                break;
+            }
+            try {
+                if (userInput.equalsIgnoreCase("bye")) {
```
you can try switch -case to make the code more neats

namprice227 commented on src/main/java/Veronica.java:
```diff
@@ -0,0 +1,152 @@
+                    System.out.println("Veronica: Bye. Hope to see you again soon! ");
+                    break;
+                } else if (userInput.equalsIgnoreCase("list")) {
+                    System.out.println("     _________________________________________________________________________________________________");
```
How do you know that this line is the perfect size, because I think the screen varies from size to size. Perhaps, there is a better way to do this?

namprice227 commented on src/main/java/Veronica.java:
```diff
@@ -0,0 +1,152 @@
+                    } catch (NumberFormatException e) {
+                        throw new VeronicaException("UHOH! Please enter a valid number after 'done'.");
+                    }
+                } else if (userInput.startsWith("unmark ")) {
```
you can take the first character of input and trim it to compare, maybe it is faster

namprice227 commented on src/main/java/Veronica.java:
```diff
@@ -0,0 +1,152 @@
+                } else if (userInput.startsWith("remove ")) { // Remove task
+                    try {
+                        int taskIndex = Integer.parseInt(userInput.substring(7)) - 1;
+                        if (taskIndex >= 0 && taskIndex < taskCount) {
```
It is good that you check the validation, but I think it is better to have a separate class to handle

namprice227 commented on src/main/java/Veronica.java:
```diff
@@ -0,0 +1,152 @@
+
+                    if (parts.length == 3) {
+                        tasks[taskCount++] = new Event(parts[0], parts[1], parts[2]);
+                        System.out.println("     _________________________________________________________________________________________________");
```
you can use the """ line to avoid print so many line

namprice227 reviewed (commented):
Overall you have good code standard, sometimes you can make the code shorter and maybe the handle can be better. Good luck =)

0x4F776C commented on src/main/java/Deadline.java:
```diff
@@ -0,0 +1,13 @@
+public class Deadline extends Task {
```
LGTM! This class follows recommended coding standard.

0x4F776C commented on src/main/java/Task.java:
```diff
@@ -0,0 +1,31 @@
+    }
+
+    public String toString() {
+        return "[" + type.name().charAt(0) + "][" + this.getStatus() + "] " + this.description;
```
Perhaps you might want to separate the statement by adding a newline before each “+”.

0x4F776C commented on src/main/java/TaskType.java:
```diff
@@ -0,0 +1,3 @@
+public enum TaskType {
```
LGTM! Short and concise!

0x4F776C commented on src/main/java/Veronica.java:
```diff
@@ -0,0 +1,152 @@
+public class Veronica {
+    public static void main(String[] args) {
+        String greet = """
+                _________________________________________________________________________________________________
```
Good work with the decluttering of long statements!

0x4F776C commented on src/main/java/Veronica.java:
```diff
@@ -0,0 +1,152 @@
+import java.util.Scanner;
+
+public class Veronica {
```
Perhaps you might want to split this file into different sub-files

* commands
* interactions

0x4F776C reviewed (commented):
LGTM overall! Might want to consider splitting your main java file to reduce clutter in a long run 😄

Melvan-Chan committed:
Add history system to save task list

Melvan-Chan committed:
Add "remove all" function to the removal of task system

Melvan-Chan committed:
Work on lvl 8 in progress

Melvan-Chan committed:
Add date and time with proper formatting

Melvan-Chan committed:
Remove task test

Melvan-Chan committed:
Refactor code for more OOP
Add Parser, TaskManager and Ui

Melvan-Chan committed:
Merge branch 'master' into add-gradle-support

Melvan-Chan committed:
Organize classes into suitable java packages

Melvan-Chan committed:
Merge pull request #1 from Melvan-Chan/add-gradle-support

Add gradle support

Melvan-Chan committed:
Add JUnit tests to test the behavior of Parser and Deadline

Melvan-Chan committed:
Package the app as an executable JAR file

Melvan-Chan committed:
Replace if-else calls in parser. Fixed indentation of switch case.

Melvan-Chan committed:
Add JavaDoc comments

Melvan-Chan committed:
Update code to fit coding standard

Melvan-Chan committed:
Add search task feature

Melvan-Chan committed:
Merge branch 'branch-Level-9' into branch-A-JavaDoc

Melvan-Chan committed:
Merge branch 'branch-A-JavaDoc' into branch-A-CodingStandard

Melvan-Chan committed:
Download and use checkStyle to detect coding style violations

Melvan-Chan committed:
Improve the code quality

Melvan-Chan committed:
Add GUI for Level-10, but still work in progress

Melvan-Chan committed:
Complete GUI for Level-10

Melvan-Chan committed:
Use var-args in UI.java

Melvan-Chan committed:
Add JavaFX UI for Veronica

This commit integrates a JavaFX-based graphical user interface (GUI)
for Veronica, replacing the previous text-based interface. The new UI
provides a more user-friendly experience with interactive elements such
as buttons, text fields, and dynamically updated task displays.

Key changes:
- Implemented JavaFX main application window.
- Added controllers for handling user input and UI updates.
- Updated task display to reflect real-time changes.
- Refactored existing CLI-based methods to work with JavaFX components.

Melvan-Chan committed:
Add assertions to verify conditions in Veronica

Added assertion checks to various methods in Veronica to ensure that
key conditions hold true during runtime.

The assertions help identify potential errors during development,
such as invalid inputs or invalid object states.

This change aims to improve code reliability and provide early detection
of logical errors, especially during debugging.

Melvan-Chan committed:
Improve code quality

Melvan-Chan committed:
Revert "Improve code quality"

This reverts commit f2c71e0d151a5e32694bab12810fdbeab3845609.

Melvan-Chan committed:
Revert "Revert "Improve code quality""

This reverts commit aa29f15ff08ba50b07e3b3fd9851d9c9b846d92b.

Melvan-Chan committed:
Commit to the proper branch

Melvan-Chan committed:
Use streams in TaskManager

Melvan-Chan committed:
Add the ability to sort items chronologically

The task manager now includes a sorting feature that orders tasks by
type (ToDo, Deadline, Event) and due date,ensuring tasks are displayed
in an organized manner.

Melvan-Chan committed:
Add Veronica User Guide and UI screenshots

Created a comprehensive User Guide for Veronica,
documenting its functionality and usage.

Added UI screenshots to enhance user understanding and navigation.

Melvan-Chan committed:
Merge branch 'master' into branch-A-Assertions

# Conflicts:
#	data/veronica_tasks.txt
#	src/main/java/veronica/task/TaskManager.java

Melvan-Chan committed:
Merge branch 'branch-A-Assertions' into branch-A-CodeQuality

# Conflicts:
#	src/main/java/veronica/main/Veronica.java
#	src/main/java/veronica/task/TaskManager.java

Melvan-Chan committed:
Fix errors and recompile with Java 17
